packages/page-broker/src/Overview/index.tsx

```diff
@@ -1,102 +1,98 @@
+    const processWorkload = (data: CoreWorkload[]) => data
+      .filter((load) => load.core === coreNumber)
+      .map((one) => (formatDataObject(one, leaseMap, reservationMap)));
+
+    const processWorkplan = (data: CoreWorkplan[]) => data
+      .filter((load) => load.core === coreNumber)
+      .map((one) => ({
+        ...formatDataObject(one, leaseMap, reservationMap),
+        timeslice: one.timeslice
+      }));
+
+    const coreData = {
+      core: coreNumber,
+      workload: workload?.length ? processWorkload(workload) : [],
+      workplan: workplan?.length ? processWorkplan(workplan) : []
     };
 
-    if ('timeslice' in w) {
-      return result as CoreWorkplanType;
-    } else {
-      return result as CoreWorkloadType;
-    }
+    return coreData;
   });
 };
 
-function Overview ({ api, apiEndpoint, className, isReady }: Props): React.ReactElement<Props> {
-  const [workLoad, setWorkLoad] = useState<CoreWorkloadType[] | undefined>();
-  const [workPlan, setWorkPlan] = useState<CoreWorkplanType[]>();
-  const [filtered, setFiltered] = useState<CoreWorkloadType[]>();
-  const { isApiReady } = useApi();
+function Overview ({ className }: Props): React.ReactElement<Props> {
+  const { api, apiEndpoint, isApiReady } = useApi();
+  const [data, setData] = useState<CoreInfo[]>([]);
+
+  const [filtered, setFiltered] = useState<CoreInfo[]>();
+  const coreCount = useBrokerStatus('coreCount') || '-';
 
-  const status = useCall<PalletBrokerStatusRecord>(isReady && api.query.broker?.status);
   const workloadInfos: CoreWorkload[] | undefined = useWorkloadInfos(api, isApiReady);
   const workplanInfos: CoreWorkplan[] | undefined = useWorkplanInfos(api, isApiReady);
   const reservations: Reservation[] | undefined = useBrokerReservations(api, isApiReady);
   const leases: LegacyLease[] | undefined = useBrokerLeases(api, isApiReady);
 
-  const leaseMap = useMemo(() => leases ? createTaskMap(leases) : [], [leases]);
-  const reservationMap = useMemo(() => reservations ? createTaskMap(reservations) : [], [reservations]);
-
-  const timesliceAsString = useMemo(() => {
-    if (!!status && !!status?.toHuman()) {
-      const timeslice = status?.toHuman().lastCommittedTimeslice?.toString();
-
-      return timeslice === undefined ? '' : timeslice.toString().split(',').join('');
-    }
-
-    return '0';
-  }, [status]);
-
-  useEffect(() =>
-    setWorkPlan(formatLoad(workplanInfos, leaseMap, reservationMap))
-  , [workplanInfos, leaseMap, reservationMap]);
+  const leaseMap: LeaseMapType = useMemo(() => leases ? createTaskMap(leases) : [], [leases]);
+  const reservationMap: ReservationMapType = useMemo(() => reservations ? createTaskMap(reservations) : [], [reservations]);
 
   useEffect(() => {
-    setWorkLoad(formatLoad(workloadInfos, leaseMap, reservationMap));
-  }, [workloadInfos, leaseMap, reservationMap]);
+    !!workplanInfos && !!workloadInfos && !!coreCount && !!leaseMap && !!reservationMap &&
```

Comment:
I dont think we need the double bangs on any of these values?

Reply:
Typescript begs to differ (for some of them) 😊 I might have thrown in some extras just in case 😅